### ISSUE TITLE
any2stringadd implicit should be removed from Predef

### ISSUE THREAD

**Issue by scabug** [scala/bug#194]
This implicit removes too much type safety, and can lead to confusing results.

I think this is more important than the '+' operator being symmetric.

*UPDATE 2015-07-29*:
**Workaround**: You can opt-out on a per-file basis by unimporting it.
```scala
import Predef.{any2stringadd => _,_}
object foo extends App{
  println(new java.util.Date + " sfsdf") // error: value + is not a member of java.util.Date
}
```

**Comment by Jasper-M** [comment]
[Another newbie confused by `any2stringadd`.](https://users.scala-lang.org/t/placeholder-syntax-not-working-while-uing-println/1715?u=jasper-m) 
While we're at it can we also remove the `def +(any: Any): String` from class `String` and replace it with `def +(str: String): String`? I guess that method is a compiler trick anyway so we can do with it whatever we want.

**Comment by SethTisue** [comment]
https://github.com/scala/scala/pull/6315, which deprecates `any2stringadd`, has been merged for 2.13.0-M5
